### PR TITLE
Add integration test for composable fully_shard and checkpoint

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #90079
* #90078
* __->__ #90041

### CODE REVIEW

File: test/distributed/_composable/test_compose.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            combo_model.seq, use_reentrant=use_reentrant
+        )
+        combo_model.seq = fully_shard(
+            combo_model.seq,
+            policy=ModuleWrapPolicy({nn.Linear}),
+        )
+
+        x = torch.randn(2, 100, device=device)
+
+        for _ in range(5):
```
Comment (awgu) [inline test/distributed/_composable/test_compose.py:75]: Might be worth testing `zero_grad(set_to_none=True)` as well:
```
for i in range(5):
    ...
    combo_optim.zero_grad(set_to_none=(i % 2 == 0))
    ...
    local_optim.zero_grad(set_to_none(i % 2 == 0))
```

**Review by awgu — APPROVED** [review]
Awesome!

File: test/distributed/_composable/test_compose.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        return self.l2(self.seq(self.l1(x)))
+
+
+class TestFSDPCheckpoint(FSDPTest):
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    @skip_if_lt_x_gpu(2)
+    @parametrize("use_reentrant", [True, False])
```
Comment (rohan-varma) [inline test/distributed/_composable/test_compose.py:53]: nit: using `for use_reentrant in [True, False]` and then `self.subTest` might be faster test wise, as it avoids spawning extra process for every parametrized test.

**Review by rohan-varma — APPROVED** [review]
LGTM, excited to see the integration between functional AC and FSDP!

File: test/distributed/_composable/test_compose.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import copy
```
Comment (awgu) [inline test/distributed/_composable/test_compose.py:1]: To fix lint:
```suggestion
# Owner(s): ["oncall: distributed"]

import copy
```

**Pull request closed without merging**